### ISSUE TITLE
[SW build] bazel 

### ISSUE THREAD

**Issue by Adam11072000** [lowRISC/opentitan#24028]
### Description

Hey,

im having trouble activating bazel to run basic tests.
it says connection timeout for some reason.
INFO: Repository io_bazel_rules_go instantiated at:
  /home/adamg/Desktop/he-soc/hardware/ip_list/opentitan/WORKSPACE:27:9: in <toplevel>
  /home/adamg/Desktop/he-soc/hardware/ip_list/opentitan/third_party/go/repos.bzl:8:17: in go_repos
Repository rule http_archive defined at:
  /root/.cache/bazel/_bazel_root/7eef5b0edb59371d5c67363668940f71/external/bazel_tools/tools/build_defs/repo/http.bzl:372:31: in <toplevel>
INFO: Repository bazel_gazelle instantiated at:
  /home/adamg/Desktop/he-soc/hardware/ip_list/opentitan/WORKSPACE:27:9: in <toplevel>
  /home/adamg/Desktop/he-soc/hardware/ip_list/opentitan/third_party/go/repos.bzl:13:17: in go_repos
Repository rule http_archive defined at:
  /root/.cache/bazel/_bazel_root/7eef5b0edb59371d5c67363668940f71/external/bazel_tools/tools/build_defs/repo/http.bzl:372:31: in <toplevel>
WARNING: Download from https://github.com/bazelbuild/rules_go/releases/download/v0.30.0/rules_go-v0.30.0.zip failed: class java.io.IOException connect timed out
ERROR: An error occurred during the fetch of repository 'io_bazel_rules_go':
   Traceback (most recent call last):
	File "/root/.cache/bazel/_bazel_root/7eef5b0edb59371d5c67363668940f71/external/bazel_tools/tools/build_defs/repo/http.bzl", line 132, column 45, in _http_archive_impl
		download_info = ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error downloading [https://github.com/bazelbuild/rules_go/releases/download/v0.30.0/rules_go-v0.30.0.zip] to /root/.cache/bazel/_bazel_root/7eef5b0edb59371d5c67363668940f71/external/io_bazel_rules_go/temp18001876725997471187/rules_go-v0.30.0.zip: connect timed out
ERROR: /home/adamg/Desktop/he-soc/hardware/ip_list/opentitan/WORKSPACE:27:9: fetching http_archive rule //external:io_bazel_rules_go: Traceback (most recent call last):
	File "/root/.cache/bazel/_bazel_root/7eef5b0edb59371d5c67363668940f71/external/bazel_tools/tools/build_defs/repo/http.bzl", line 132, column 45, in _http_archive_impl
		download_info = ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error downloading [https://github.com/bazelbuild/rules_go/releases/download/v0.30.0/rules_go-v0.30.0.zip] to /root/.cache/bazel/_bazel_root/7eef5b0edb59371d5c67363668940f71/external/io_bazel_rules_go/temp18001876725997471187/rules_go-v0.30.0.zip: connect timed out
ERROR: Error computing the main repository mapping: no such package '@io_bazel_rules_go//go': java.io.IOException: Error downloading [https://github.com/bazelbuild/rules_go/releases/download/v0.30.0/rules_go-v0.30.0.zip] to /root/.cache/bazel/_bazel_root/7eef5b0edb59371d5c67363668940f71/external/io_bazel_rules_go/temp18001876725997471187/rules_go-v0.30.0.zip: connect timed out
cp: cannot stat '/home/adamg/Desktop/he-soc/hardware/ip_list/opentitan/bazel-out/k8-fastbuild-ST-2cc462681f62/bin//sw/tests/opentitan/sram_aes_smoketest/sram_aes_smoketest_sim_verilator.elf': No such file or directory
cp: cannot stat '/home/adamg/Desktop/he-soc/hardware/ip_list/opentitan/bazel-out/k8-fastbuild-ST-2cc462681f62/bin//sw/tests/opentitan/sram_aes_smoketest/sram_aes_smoketest_sim_verilator.dis': No such file or directory
make: Nothing to be done for 'compile-bazel-sram'.

all the python reqs and the apt reqs have been downloaded.

any idea?

thanks

**Comment by Adam11072000** [comment]
well, this shot up as a firewall problem. although now i do have another problem, regarding the build of bazel rules:
Saved /home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/ot_python_deps/zipfile2-0.0.12-py2.py3-none-any.whl
 (Traceback (most recent call last):
  File "/home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/python3_x86_64-unknown-linux-gnu/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/python3_x86_64-unknown-linux-gnu/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/rules_python/python/pip_install/extract_wheels/__main__.py", line 5, in <module>
    main()
  File "/home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/rules_python/python/pip_install/extract_wheels/__init__.py", line 117, in main
    targets = [
  File "/home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/rules_python/python/pip_install/extract_wheels/__init__.py", line 120, in <listcomp>
    bazel.extract_wheel(
  File "/home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/rules_python/python/pip_install/extract_wheels/lib/bazel.py", line 412, in extract_wheel
    for name, entry_point in sorted(whl.entry_points().items()):
  File "/home/adam/.cache/bazel/_bazel_adam/60daae82b1d411e51ac71d4764045752/external/rules_python/python/pip_install/extract_wheels/lib/wheel.py", line 57, in entry_points
    name = "{}-{}".format(metadata.name.replace("-", "_"), metadata.version)
AttributeError: 'NoneType' object has no attribute 'replace'
)
ERROR: Analysis of target '//sw/tests/opentitan/sram_aes_smoketest:sram_aes_smoketest' failed; build aborted: 
INFO: Elapsed time: 63.237s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (104 packages loaded, 2376 targets configured)
    currently loading: util

**Comment by jwnrt** [comment]
Hi @Adam11072000, I've had this same issue recently

The problem was that Bazel's `rules_python` was internally using the `pkginfo` Python package installed on my system and not inside Bazel, and that `pkginfo` is version 1.9.6. This version does not have support for metadata version 2.3 ([which was added in the next release](https://pypi.org/project/pkginfo/1.10.0/)).

Some of the packages that `rules_python` is trying to install have metadata version 2.3 where the `metadata` does not have a `name` field, causing that issue.

Are you building the latest version of OpenTitan? I only hit this error on an old commit from before we moved to having a generated `python-requirements.txt` that contains hashes of all our packages.

On that old commit the packages with metadata version 2.3 were implicit, so I worked around the issue by explicitly specifying older versions of those packages in `python-requirements.txt` that had older metadata versions.

To fix this properly, I think we want to explicitly specify the `pkginfo` that Bazel's `rules_python` uses to build dependencies. I'm not sure exactly how to do that but will try have a look.

**Comment by Adam11072000** [comment]
hey,
no this is not the latest commit. this is a forked repo used by El-Saqr.
they did not change the bazel build system, but rather added some makefiles to allow building using questa too.
so essentially bazel should work.

**Comment by jwnrt** [comment]
Unfortunately I think it requires a change to Bazel itself to make `pkginfo` work properly.
The version is fixed here: https://github.com/bazelbuild/rules_python/blob/bf704297d85cdcf2338715a6f8411e49a679da16/tools/publish/requirements.txt#L245

Instead, please could you try this workaround by adding the following lines to `python-requirements.txt`?

```
urllib3==1.26.19
pygments==2.13.0
```

**Comment by Adam11072000** [comment]
same issue.
im dropping the require hashes part due to this failure.
Collecting git+https://github.com/lowRISC/fusesoc.git@ot-0.3 (from -r python-requirements.txt (line 52))
ERROR: Can't verify hashes for these requirements because we don't have a way to hash version control repositories:
    git+https://github.com/lowRISC/fusesoc.git@ot-0.3 (from -r python-requirements.txt (line 52))

i am running this command to install the reqs:
 pip3 install --user -r python-requirements.txt

**Comment by jwnrt** [comment]
Please could you try:

```sh
pip install --user pkginfo==1.10.0
```

**Comment by Adam11072000** [comment]
i ran the pip3 install and also reinstalled python reqs,
still same error.

**Comment by jwnrt** [comment]
In that case I'm not sure how to work around the error in that fork, sorry. They may need to update their copy of OpenTitan to fix the issue.

**Comment by jwnrt** [comment]
I’m going to close this since I don’t think there’s much we can do to fix the forked repo. I’m sorry we couldn’t help in this case.

**Comment by pqcfox** [comment]
Following up on this, I ran into the same exact issue as well. Applying the fix in https://github.com/lowRISC/rules_python/pull/2 to my Bazel's output base resolved the issue, since `pkginfo` 1.12 can handle through metadata version 2.4, which the most recent `setuptools` generates upon fetching source distributions.

Once that PR is merged, the problem should be resolved for good :)
